### PR TITLE
Overhaul Search control with ranked results and improved navigation

### PR DESCRIPTION
**Pull request by brandonxu360**
This pull request enhances the `Search` component and its test suite by introducing support for searching through watershed data, improving coordinate validation, and expanding test coverage for different search scenarios. The changes primarily focus on passing watershed data into the search control, updating the test setup, and adding tests for watershed and coordinate search behaviors.

**Key changes include:**

**Search functionality and integration:**
- The `SearchControl` component is now passed a `watersheds` prop from `WatershedMap`, enabling it to search and suggest watershed features.

**Test suite improvements and new test data:**
- The `Search` component tests are refactored to accept a `watersheds` prop, and a helper (`makeWatershedProps`) plus a `testWatersheds` GeoJSON fixture are added to simulate realistic search scenarios.
- Mocks for navigation and context hooks are introduced to simulate routing and context-dependent behavior in tests.

**Expanded search behavior test coverage:**
- New tests verify:
  - Navigating home before leaving the search context when searching by coordinates from a watershed route.
  - Proper error handling for invalid and out-of-range coordinates, including user feedback via toasts.
  - Watershed search suggestions, including unique and multiple matches, and handling of no matches.
  - Highlighting of matched attributes/values (e.g., `sourceName`) in search results.
This pull request enhances the test coverage and robustness of the `Search` component in the map interface, especially for watershed search functionality. It introduces a realistic set of test watersheds, mocks relevant hooks and navigation, and adds comprehensive tests for searching by coordinates and watershed names, including edge cases and error handling.

**Testing and Mocking Improvements:**

* Added a `testWatersheds` GeoJSON dataset and a utility function `makeWatershedProps` to provide realistic watershed data for tests.
* Mocked `useNavigate` from `@tanstack/react-router` and `useRunId` hook to simulate navigation and context-dependent behavior in tests.

**Expanded Search Component Test Coverage:**

* Added tests for coordinate search from a watershed context, ensuring navigation to home before leaving the current watershed.
* Added tests for invalid and out-of-range coordinate input, verifying correct error toast messages and modal behavior. [[1]](diffhunk://#diff-f7b58dff36820e58ae8d3bb9d3dd59f201093168ef07635fab55a4d1a0235e1cL78-R230) [[2]](diffhunk://#diff-f7b58dff36820e58ae8d3bb9d3dd59f201093168ef07635fab55a4d1a0235e1cR243-R325)
* Added tests for watershed name search, including unique and multiple matches, no matches, and attribute highlighting in results.

**Component Prop Update:**

* Updated `WatershedMap` to pass the `watersheds` prop to `SearchControl`, ensuring the search component receives necessary data.

### CODE REVIEW

File: client/src/components/map/controls/Search.tsx [hunk 1]
```diff
@@ -1,10 +1,271 @@
...
+  return lat >= -90 && lat <= 90 && lng >= -180 && lng <= 180;
+}
+
+function isCoordinateLike(input: string): boolean {
+  const normalized = input.trim();
+  return (
+    /\d/.test(normalized) &&
+    (normalized.includes(",") ||
+      normalized.split(/\s+/).filter(Boolean).length >= 2)
+  );
```
[the remark this answers is not in the document]
Comment (brandonxu360) [reply client/src/components/map/controls/Search.tsx]: Updated the heuristic in 67c090c (requires two numeric tokens unless parsing succeeds or a comma suggests a coord attempt) and added a test so names like “Deer Creek 2” still watershed-search.

Still pretty naive, may consider removing coordinate search entirely.

File: client/src/components/map/controls/Search.tsx [hunk 2]
```diff
@@ -34,81 +295,226 @@ const useStyles = tss.create(({ theme }) => ({
...
+      if (runId) {
+        navigate({ to: "/" });
+      }
+      closeAndReset();
+      return;
+    }
+
+    if (isCoordinateLike(trimmedInput)) {
       toast.error(
         'Invalid coordinates. Please enter in "latitude, longitude" format.',
```
[the remark this answers is not in the document]
Comment (brandonxu360) [reply client/src/components/map/controls/Search.tsx]: Correct - the parser already accepts comma- or space-separated pairs. Toast copy is aligned in 79334e2.

Could be changed to be more strict instead so that less conflicts between attempted watershed searches and coordinate searches occur.

File: client/src/components/map/controls/Search.tsx [hunk 3]
```diff
@@ -34,81 +295,226 @@ const useStyles = tss.create(({ theme }) => ({
...
       );
+      return;
+    }
+
+    const matches = rankWatershedMatches(normalizedQuery, searchIndex);
+    if (matches.length === 0) {
+      setResults([]);
+      toast.error("No watershed match found. Try coordinates or another name.");
+      return;
```
[the remark this answers is not in the document]
Comment (brandonxu360) [reply client/src/components/map/controls/Search.tsx:508]: When there’s no searchable watershed data we now show a loading/unavailable message instead of “no match,” with coverage in b8e32a6.

File: client/src/components/map/controls/Search.tsx [hunk 4]
```diff
@@ -34,81 +295,226 @@ const useStyles = tss.create(({ theme }) => ({
...
-    padding: theme.spacing(1),
+    background: theme.palette.background.paper,
+    color: theme.palette.text.primary,
+    padding: theme.spacing(1.25),
     borderRadius: theme.spacing(1),
-    boxShadow: "0 2px 10px rgba(0, 0, 0, 0.5)",
+    border: `1px solid ${theme.palette.surface.border}`,
+    boxShadow: "0 10px 26px rgba(0, 0, 0, 0.35)",
     zIndex: 1000,
+    minWidth: 400,
```
[the remark this answers is not in the document]
Comment (brandonxu360) [reply client/src/components/map/controls/Search.tsx]: Not really a priority since we currently don't allow narrow viewports to begin with, but replaced the fixed min width with a responsive width in dadff31 so the panel can shrink on small screens anyways.
